src/libcore/option.rs

```diff
@@ -913,6 +913,65 @@ impl<T> Option<T> {
     pub fn replace(&mut self, value: T) -> Option<T> {
         mem::replace(self, Some(value))
     }
+
+    /// Zips `self` with another `Option`.
+    ///
+    /// Returns `Some((_, _))` when both `self` and `other`
```

Comment:
"when" -> "if". "when" in English is mainly about time, not condition. At least as far as I -- a non native speaker -- can tell :P If a native could confirm this, that'd be great.